deployment/helm/templates/pre-delete-job.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if .Values.cleanupHook.enabled }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ .Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        app.kubernetes.io/component: helm-hook
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: delete-configurations
+        image: curlimages/curl:latest
+        command: ["/bin/sh", "-c"]
+        args: [
+          "curl -ik -X DELETE -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" https://kubernetes.default.svc.cluster.local/apis/akri.sh/v0/namespaces/default/configurations"
```

Comment:
Question: will this `https://kubernetes.default.svc.cluster.local/apis/akri.sh/v0/namespaces/default/configurations` work even if _akri_ is installed in a namespace other than _default_?

Reply:
this was one of the tests i wanted to run. I confirmed your concern. We need to template the namespace in. I updated it and am able to confirm it cleans up namespace installations

Reply:
The Configurations can also be scattered over multiple namespaces. 
We should do the equivalent of `kubectl delete -A --all --wait akric` (or maybe just use kubectl here rather than crafting the curl request)

Reply:
The fact that Akri handles configs outside it's namespace is weird behavior to me. If i install Akri in the `foo` ns and then apply a config to the `bar` ns, the config is seen by the `foo` agent and instances are created and broker pods deployed (both in the `bar` namespace):
```sh
foo           pod/akri-agent-daemonset-2j2tc                                 1/1     Running     0          2m46s
foo           pod/akri-agent-daemonset-fk8pm                                 1/1     Running     0          2m46s
foo           pod/akri-agent-daemonset-89mhp                                 1/1     Running     0          2m46s
foo           pod/akri-controller-deployment-695fcd4c4f-mf6js                1/1     Running     0          2m46s
foo           pod/akri-webhook-configuration-6b6947dbc-qjwpj                 1/1     Running     0          2m46s
bar           pod/k3d-test-cluster-server-0-akri-debug-echo-foo-a19705-pod   1/1     Running     0          110s
bar           pod/k3d-test-cluster-server-0-akri-debug-echo-foo-8120fe-pod   1/1     Running     0          110s
bar           pod/k3d-test-cluster-agent-1-akri-debug-echo-foo-a19705-pod    1/1     Running     0          110s
bar           pod/k3d-test-cluster-agent-1-akri-debug-echo-foo-8120fe-pod    1/1     Running     0          110s
bar           pod/k3d-test-cluster-agent-0-akri-debug-echo-foo-8120fe-pod    1/1     Running     0          110s
bar           pod/k3d-test-cluster-agent-0-akri-debug-echo-foo-a19705-pod    1/1     Running     0          110s

NAMESPACE   NAME                                          CONFIG                SHARED   NODES                                                                                 AGE
bar         instance.akri.sh/akri-debug-echo-foo-a19705   akri-debug-echo-foo   true     ["k3d-test-cluster-agent-0","k3d-test-cluster-server-0","k3d-test-cluster-agent-1"]   110s
bar         instance.akri.sh/akri-debug-echo-foo-8120fe   akri-debug-echo-foo   true     ["k3d-test-cluster-agent-0","k3d-test-cluster-server-0","k3d-test-cluster-agent-1"]   110s

NAMESPACE   NAME                                        CAPACITY   AGE
bar         configuration.akri.sh/akri-debug-echo-foo   5          110s
```
I would expect the agent to only watch resources in it's ns? That is clearly not the behavior. Is that an issue we want to look at? If not, then yes, it makes sense to delete configs across all namespaces

Reply:
The plan for splitting the Configuration resource is to have the discovery part to be cluster wide, and only the broker part be namespaced, so it will probably be clearer when done.

But, as is, I expect akri to watch for resources across all namespaces, as I usually want the broker pods to be in app specific namespaces that are not the same as the akri namespace (when doing tests, I usually install akri in its own namespace and creates my Configurations in the default one)

Reply:
I may not have time to adjust this to be cluster wide before we release tomorrow. I will try to get to it, but if not, would we be interested in merging this as is?

Reply:
Updated to use `kubectl`